quantum/keyboard.c

```diff
@@ -627,27 +638,40 @@ void keyboard_task(void) {
     }
 #endif
 
+#ifdef POINTING_DEVICE_ENABLE
+    pointing_device_task();
+    const bool pointing_device_changed = !(last_pointing_device_activity_elapsed());
+#endif
```

Comment:
Not sure this is the best way to handle this. 

but short of significantly changing pointing device code, or tracking the changed state outside of the keyboard task loop...

Reply:
That was may main concern, as well. 

I thought about changing pointing device task and send to be bool, but that would be a much more significant and invasive change.

Reply:
Phase 2, if this PR's implementation is found to be suboptimal?

Reply:
> Phase 2, if this PR's implementation is found to be suboptimal?

Rewrite the task and send functions.  Make send only handle sending the report (and zeroing out the vector movement), and let the task handle the send... 

You know what, screw it... doing that now.

Reply:
rewrote anyways.  This should handle setting the reports outside of the pointing device task, and still properly report activity.